apps/addons/models.py

```diff
@@ -497,7 +494,14 @@ def delete(self, msg='', reason=''):
     @classmethod
     def initialize_addon_from_upload(cls, data, is_listed=True):
         fields = cls._meta.get_all_field_names()
+        # Reclaim GUID from deleted add-on.
+        try:
+            old_guid_addon = Addon.unfiltered.get(guid=data['guid'])
+            old_guid_addon.update(guid=None)
```

Comment:
guid is supposed to be a unique value in the model, so I didn't want to cause any database errors.  I can change if its okay to ignore the unique-ness defined in the model.